facade/src/main/java/org/jboss/pnc/facade/providers/ArtifactProvider.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014-2019 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.facade.providers;
+
+import org.jboss.pnc.dto.ArtifactRef;
+import org.jboss.pnc.dto.response.Page;
+import org.jboss.pnc.facade.mapper.api.ArtifactMapper;
+import org.jboss.pnc.facade.rsql.mapper.ArtifactRSQLMapper;
+import org.jboss.pnc.facade.validation.DTOValidationException;
+import org.jboss.pnc.model.Artifact;
+import org.jboss.pnc.spi.datastore.repositories.ArtifactRepository;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import java.util.Optional;
+
+import static org.jboss.pnc.spi.datastore.predicates.ArtifactPredicates.withMd5;
+import static org.jboss.pnc.spi.datastore.predicates.ArtifactPredicates.withSha1;
+import static org.jboss.pnc.spi.datastore.predicates.ArtifactPredicates.withSha256;
+/**
+ * @author Jan Michalov <jmichalo@redhat.com>
+ */
+@Stateless
+public class ArtifactProvider extends AbstractProvider<Artifact, org.jboss.pnc.dto.Artifact, ArtifactRef>{
+
+    @Deprecated
+    public ArtifactProvider() {
+    }
```

Comment:
Instead of using public (!) deprecated noarg constructor, create an interface for the ArtifactProvider.

See https://docs.jboss.org/weld/reference/latest/en-US/html/injection.html#_client_proxies

Reply:
Didn't know :) thanks for that